blocks/basic/include/gnuradio-4.0/basic/FunctionGenerator.hpp

```diff
@@ -0,0 +1,297 @@
+Users can create sequences of various functions using tags.
+These tags are generated through dedicated helper methods, named `create{FunctionName}Tag(...)`.
+This method enables the flexible assembly of diverse function chains, offering significant versatility in signal generation.
+
+Supported function types and their configurable parameters:
+
+* Const
+  - Constantly returns the same value, specified by 'startValue'.
+  To create a `property_map` use `createConstPropertyMap(tagIndex, startValue)`.
+
+* LinearRamp
+  - Interpolates linearly between a start and a final value over a set duration.
+  To create a `property_map` use `createLinearRampPropertyMap(tagIndex, startValue, finalValue, duration)`.
+
+* ParabolicRamp
+  - Produces a function with three sections: parabolic-linear-parabolic. The 'roundOffTime' parameter determines the length of the parabolic sections.
+  To create a `property_map` use `createParabolicRampPropertyMap(tagIndex, startValue, finalValue, duration, roundOffTime)`.
+
+* CubicSpline
+  - Implements smooth cubic spline interpolation between start and final values over a specified duration.
+  To create a `property_map` use `createCubicSplinePropertyMap(tagIndex, startValue, finalValue, duration)`.
+
+* ImpulseResponse
+  - Creates a function that spikes from 'time0' for a duration of 'time1'.
+  To create a `property_map` use `createImpulseResponsePropertyMap(tagIndex, startValue, finalValue, time0, time1)`.
+
+To create a chain of functions one can use `ClockSource`. There are 2 ways to do it:
+1) Using times + property_map
+Firstly, add time Tag entries, with time and command string:
+@code
+auto &clockSrc = testGraph.emplaceBlock<gr::basic::ClockSource<float>>({ { "sample_rate", 100.f }, { "n_samples_max", 100 }, { "name", "ClockSource" } });
+clockSrc.addTimeTagEntry(1'000, "CMD_BP_START:FAIR.SELECTOR.C=1:S=1:P=1");
+clockSrc.addTimeTagEntry(10'000, "CMD_BP_START:FAIR.SELECTOR.C=1:S=1:P=2");
+clockSrc.addTimeTagEntry(30'000, "CMD_BP_START:FAIR.SELECTOR.C=1:S=1:P=3");
+@endcode
+
+Secondly, create table to match command and functions:
+@code
+funcGen.addFunctionTableEntry({ { "context", "CMD_BP_START:FAIR.SELECTOR.C=1:S=1:P=1" } }, createConstPropertyMap(5.f));
+funcGen.addFunctionTableEntry({ { "context", "CMD_BP_START:FAIR.SELECTOR.C=1:S=1:P=2" } }, createLinearRampPropertyMap(5.f, 30.f, .2f));
+funcGen.addFunctionTableEntry({ { "context", "CMD_BP_START:FAIR.SELECTOR.C=1:S=1:P=3" } }, createConstPropertyMap(30.f));
+@endcode
+
+Lastly create a matching predicate with the following signature:
+@code
+std::function<std::optional<bool>(const property_map &, const property_map &, std::size_t)>;
+@endcode
+
+
+2) Using user defined ready-to-use Tags.
+@code
+auto &clockSrc = testGraph.emplaceBlock<gr::basic::ClockSource<float>>({ { "sample_rate", 100.f }, { "n_samples_max", 100 }, { "name", "ClockSource" } });
```

Comment:
minor (and also this is just within a code documentation): maybe we should re-use the `sample_rate` etc. definitions from the default tag definitions to indicate that these are also user-level API definitions and conventions.

Reply:
For the moment there is a problem with `gr::` prefix, this first need to be fixed. And then one should use where possible definitions from default tags.